src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -36,17 +36,15 @@ public EventPublishingTests(ITestOutputHelper testOutput)
 
         public async Task DisposeAsync()
         {
-            await _hybridConnectionHost.Stop();
+            await _serviceBusEventConsumerHost.StopAsync();
         }
 
         public async Task InitializeAsync()
         {
-            var relayNamespace = Configuration.GetValue<string>("Arcus:HybridConnections:RelayNamespace");
-            var hybridConnectionName = Configuration.GetValue<string>("Arcus:HybridConnections:Name");
-            var accessPolicyName = Configuration.GetValue<string>("Arcus:HybridConnections:AccessPolicyName");
-            var accessPolicyKey = Configuration.GetValue<string>("Arcus:HybridConnections:AccessPolicyKey");
+            var connectionString = Configuration.GetValue<string>("Arcus:ServiceBus:ConnectionString");
+            var topicName = Configuration.GetValue<string>("Arcus:ServiceBus:TopicName");
 
-            _hybridConnectionHost = await HybridConnectionHost.Start(relayNamespace, hybridConnectionName, accessPolicyName, accessPolicyKey, _testLogger);
+            _serviceBusEventConsumerHost = await ServiceBusEventConsumerHost.Start(topicName, connectionString, _testLogger);
```

Comment:
I would also like to introduce a boolean that indicates if the subscription should be deleted afterwards instead of always doing it.

We could use an options object for only flag:
```suggestion
var options = new ServiceBusEventConsumerHostOptions(deleteSubscriptionOnClose: true);
_serviceBusEventConsumerHost = await ServiceBusEventConsumerHost.Start(topicName, connectionString, _testLogger, options);
```

We could use an options object for only flag and logging which also provides a default ctor that uses `NoOpLogger`.

Example:
```suggestion
var options = new ServiceBusEventConsumerHostOptions(logger, deleteSubscriptionOnClose: true);
_serviceBusEventConsumerHost = await ServiceBusEventConsumerHost.Start(topicName, connectionString, options);
```

Move service bus configuration with flag to options:
```suggestion
var options = new ServiceBusEventConsumerHostOptions(topicName, connectionString, deleteSubscriptionOnClose: true);
_serviceBusEventConsumerHost = await ServiceBusEventConsumerHost.Start(options, logger);
```

Last but not least, move everything to options:
```suggestion
var options = new ServiceBusEventConsumerHostOptions(topicName, connectionString, logger, deleteSubscriptionOnClose: true);
_serviceBusEventConsumerHost = await ServiceBusEventConsumerHost.Start(options);
```

Any thoughts?

Reply:
I prefer the third option. The logger is not a configuration/option, it's a _dependency_ of that class. All the other parameters are related to the actual behavior/configuration of the class.

Reply:
Good point @samneirinck, now I get what you mean. @stijnmoreels where would you use an enum for what exactly? I'm not following I think.

Reply:
Went for option 3 for now and will wait on clarification by Stijn before resolving

Reply:
Discussed this with Stijn offline and we're going to use `SubscriptionBehavior.DeleteOnClosure` & `SubscriptionBehavior.KeepOnClosure` instead of a boolean.